src/python/pants/goal/run_tracker.py

```diff
@@ -105,6 +107,55 @@ def start(self, run_start_time: float, specs: List[str]) -> None:
             }
         )
 
+    def get_anonymous_telemetry_data(self, repo_id: str) -> Dict[str, str | List[str]]:
+        # TODO: Find a way to know from a goal name whether it's a standard or a custom
+        #  goal whose name could, in theory, reveal something proprietary. That's more work than
+        #  we want to do at the moment, so we maintain this manual list for now.
+        standard_goals = {
+            "count-loc",
+            "dependees",
+            "dependencies",
+            "export-codegen",
+            "filedeps",
+            "filter",
+            "fmt",
+            "lint",
+            "list",
+            "package",
+            "py-constraints",
+            "repl",
+            "roots",
+            "run",
+            "tailor",
+            "test",
+            "typecheck",
+            "validate",
+        }
+
+        def maybe_hash_with_repo_id_prefix(s: str) -> str:
+            qualified_str = f"{repo_id}.{s}" if s else repo_id
+            # If the repo_id is the empty string we return a blank string.
+            return sha256(qualified_str.encode()).hexdigest() if repo_id else ""
+
+        return {
+            "run_id": str(self._run_info.get("id", uuid.uuid4())),
```

Comment:
seems to be that run_info always has a run_id....

Reply:
Yes, but I want to be extra bullet-proof here so that we never crash because of telemetry.